src/UI/Art/SpriteArtist/OverlayArtist.java

```diff
@@ -59,15 +59,22 @@ public static void drawHighlights(Graphics bigG,
         overlays.add(ov);
 
     // Check if input is the same as last time
-    ByteBuffer params = ByteBuffer.allocate(Integer.BYTES * (8 + inputOverlays.size()) );
+    ByteBuffer params = ByteBuffer.allocate(Integer.BYTES * (8 + inputOverlays.size()*3) );
```

Comment:
Magic 3?

Reply:
Also, magic 8?

Reply:
The magic numbers are how many integers I'm storing of each type, in this function.
Potentially 8 ints non-scaling, and potentially 3 ints per overlay that's requested for display.
I don't think defining constants or whatever will make the code much more clear?

Reply:
What are we actually calculating? The number of overlays? The number of tiles that can be overlaid? And why did you have to introduce the magic three when we didn't have it before? What quantity are we calculating? Even after reading your description I have no idea.